src/plugins/Workers/SyncWorker/initialSyncUp.js

```diff
@@ -10,23 +10,34 @@ const fetchAndStoreAddressTransactions = require('./utils/fetchAndStoreAddressTr
 // fetching here found that we have more addresses that on starting of the sync up.
 module.exports = async function initialSyncUp() {
   const { transporter, storage } = this;
-
   const addrList = this.getAddressListToSync().map((addr) => addr.address);
 
   // Due to the events system, we need to handle the fact that we did subscribed to addresses
   // that we had received the transactions and store before
   // being able to release initialSyncUp as ready.
   // When we will move to bloomfilter, that part might be more complex.
   /* eslint-disable-next-line no-async-promise-executor */
-  return new Promise(async (resolve) => {
-    // eslint-disable-next-line no-restricted-syntax
-    for (const address of addrList) {
-      // We need to wait for fetching and addition to be done sequentially
-      // to correctly manage our balance as input can be output in prev tx.
-      /* eslint-disable-next-line no-await-in-loop */
-      await fetchAndStoreAddressTransactions(address, transporter, storage);
-    }
-    logger.silly('SyncWorker - initialSyncUp - Fully synced');
-    resolve();
+  const promises = [];
+  addrList.forEach((address) => {
+    promises.push(fetchAddressTransactions(address, transporter));
+  });
+
+  let transactions = [];
+
+  const fetchedPromises = await Promise.all(promises);
+
+  fetchedPromises.forEach((fetchedTransactions) => {
+    transactions = transactions.concat(fetchedTransactions);
   });
+
+  const ordered = new TransactionOrderer();
+
+  transactions.forEach((tx) => ordered.insert(tx));
+
+  const importPromises = [];
+  ordered.transactions.forEach((orderTransaction) => {
+    importPromises.push(storage.importTransaction(orderTransaction));
+  });
+
+  return Promise.all(importPromises).then(() => logger.silly('SyncWorker - initialSyncUp - Fully synced'));
```

Comment:
Combination of `async/await` and promise callbacks doesn't look clean usually :(

I believe it will look better with await:

```
await Promise.all(importPromises);

logger.silly('SyncWorker - initialSyncUp - Fully synced');
```

Reply:
Indeed, that will improve readability, I've pushed the change.